### DISCUSSION TITLE
[React] Conditional sticky behaviour

### DISCUSSION THREAD

**Discussion by jossmac** [floating-ui/floating-ui#2316] [Q&A] [answered]
How would you implement a popover that behaves traditionally until the reference element escapes the scroll boundary, at which point the popover would "stick" to the bottom of the scroll boundary.

I'm trying to emulate something like this from Confluence:

![Large GIF (772x484)](https://user-images.githubusercontent.com/2730833/236117999-336f570f-76cb-4ea2-a3df-0f2f5bf08335.gif)

I can get the middleware data from [hide()](https://floating-ui.com/docs/hide) but not sure of the best way to approach the "sticky" implementation.

**Comment by atomiks** [comment, marked as answer]
Is that the same as the [`shift.crossAxis` demo](https://floating-ui.com/docs/shift#crossaxis)?

**Reply by jossmac** [reply]
That looks promising, thank you!

I've got something working by providing a fn to `offset()` but it feels a bit janky. I'll take another stab at it with the suggested API and let you know how it goes.

**Reply by jossmac** [reply]
Sorted. So much tidier, thanks again!
